mindsdb/integrations/handlers/xgboost_handler/xgboost_handler.py

```diff
@@ -0,0 +1,471 @@
+            log.logger.info("best_params: " + optimized_parameter_log)
+            log.logger.info("best_value: " + str(study.best_value))
+            config.update(study.best_params)
+            config.update({"eval_metric": metric_str, "early_stopping_rounds": self.DEFAULT_EARLYSTOP_ROUND})
+            if task_type == TaskType.regressor:
+                model = XGBRegressor(**config)
+            else:
+                model = XGBClassifier(**config)
+        # train model
+        log.logger.info(F"Model training.")
+        try:
+            model.fit(train_x, train_y, eval_set=[(train_x, train_y)], verbose=50)
+        except Exception as e:
+            log.logger.warn(F"Model training failed: " + self.name)
+            raise e
+        log.logger.info(F"Model trained.")
+        # save args, model, feature transformer
+        args[self.ARG_TARGET] = target
+        args[self.ARG_USING_TASK] = task_type.value
+        args[self.ARG_CLASS_NUM] = class_num
+        args[self.ARG_COLUMN_SEQUENCE] = candidate_feature_columns
+        log.logger.info(F"Model config: " + json.dumps(args))
+        model_bytes = model.get_booster().save_raw()
+        feature_transformer_bytes = pickle.dumps(feature_transformer)
+        self.model_storage.json_set(self.PERSISIT_ARGS_KEY_IN_JSON_STORAGE, args)
+        self.model_storage.file_set(self.PERSISIT_MODEL_FILE_NAME, model_bytes)
+        self.model_storage.file_set(self.PERSISIT_FEATURE_TRANSFORMER, feature_transformer_bytes)
+        log.logger.info(F"Model saved.")
+
+    def predict(self, df, args):
+        # read from storage
+        args = self.model_storage.json_get(self.PERSISIT_ARGS_KEY_IN_JSON_STORAGE)
+        feature_transofrmer_bytes = self.model_storage.file_get(self.PERSISIT_FEATURE_TRANSFORMER)
+        boost_model_bytes = self.model_storage.file_get(self.PERSISIT_MODEL_FILE_NAME)
+        # initialize args, model, transformer
+        target = args[self.ARG_TARGET]
+        task_type = TaskType(args[self.ARG_USING_TASK])
+        class_num = args[self.ARG_CLASS_NUM]
+        feature_transformer: FeatureTransformer = pickle.loads(feature_transofrmer_bytes)
+        boost_model: xgb.Booster = xgb.Booster(model_file=bytearray(boost_model_bytes))
+        # predict
+        p_feature_data = feature_transformer.transform(df=df)
+        pred_data = boost_model.predict(p_feature_data)
+        if task_type == TaskType.classifier:
+            label_arr = feature_transformer.label_encoder.inverse_transform([i for i in range(class_num)])
+            pred_label = [str(label_arr[i]) for i in np.argmax(pred_data, axis=1)]
+            pred_probs_arr = []
+            for probs in pred_data:
+                prob_items = []
+                for i, prob in enumerate(probs):
+                    prob_items.append("\"" + str(label_arr[i]) + "\"=" + str(round(prob, 5)))
+                pred_probs_arr.append("{" + ",".join(prob_items) + "}")
```

Comment:
Why is `pred_probs_arr` a list of strings? I would imagine a list of lists (with `object` dtype) to be okay for the MindsDB planner + executioner (we do this in the Lightwood handler).